ir-engine/etherealengine-archive
# fix(ir-engine): fixed corrupted projects due git http connection

## Summary
When using http connections instead of ssh connections, for some reason I faced several errors while trying to install packages using the manifest approach, figured out that directly cloning these projects within packages/projects/projects fixed all these random errors I kept getting, because of all of that trouble I faced I decided to reverse engineer the manifest approach flow and to fix the issue.

As a context I didn't face these issues while staging locally in macos since it only asked me for username and password/token once, while in wsl ubuntu it kept asking me for username and password/token for each project being installed from these manifests, for some reason the credentials reference were lost denying access to clone each project after the second clone in the iteration, in order to fix the problem I enforced SSH connection instead and now it does work as expected.

The second problem was that because of the first problem, while trying to figure out I installed one of these projects using the --branch="main" flag which caused many other errors while trying to run the project locally, since the main branch is outdated and it's supposed to install from the dev branch, but the manifest script wouldn't allow to fix the problem and instead would skip a directory that is already existing, such problem could have been manually fixed by changing the branch on each of the projects having code from an outdated branch which is unproductive and because of that I also introduced a new feature controlled by a new flag `--replace="yes"` that would allow to decide if override a current installation.

During the process of solving the root cause and improving the user experience, I found a few improvements opportunities like deprecating found antipatterns and unnecessary operations hurting performance.

This PR fixes the HTTP related problems, introduces a new useful feature and Improves performance.

## Subtasks Checklist

[x] Enforce SSH connection without breaking installations from manifests having http connections configured
[x] Introduced optional `--replace="yes"` flag that allows to reinstall a project from a different branch (Default prop value = "no")
[x] Deprecated antipatterns
[x] Improved performance

## Breaking Changes
None

## References
closes #_insert number here_

## QA Steps
- Install projects using manifest script in the exact same way is currently expected (validates no breaking changes are introduced)
- Re-install projects using new `--replace="yes"` flag using a different branch than originally installed
- Execute the command `git status` and `git branch` from within an installed project to validate that the code has been pulled from the expected branch, also to confirm that the given branch is the only branch locally.

![image](https://github.com/EtherealEngine/etherealengine/assets/1533239/84ed3fe0-60e0-4f21-a576-a8488b86d6ad)
